Problems with `Development.Module` change

**Describe the bug**
After the merge of #2195 I am not able to configure the current NEST master branch via cmake anymore.

**To Reproduce**
Python 3.9.7 (from Conda, python=3.9.7=hb7a2778_3_cpython )
Cmake 3.21.3 (from Conda, cmake=3.21.3=h8897547_0)


**Desktop/Environment (please complete the following information):**
 - OS: NAME="Ubuntu" VERSION="20.04.3 LTS (Focal Fossa)"
 - Shell: bash
 - NEST-Version: master

**Additional context**
cmake -DCMAKE_INSTALL_PREFIX:PATH=$install_dir -Dwith-python=ON -Dwith-ltdl=OFF -DCMAKE_CXX_COMPILER=g++ -DCMAKE_C_COMPILER=gcc $nest_dir
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
-- Host triple: x86_64-pc-linux
-- Target triple: x86_64-pc-linux
CMake Error at /home/kurth/miniconda3/envs/nest_dev/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Python (missing: Development.Module) (found suitable version
  "3.9.7", minimum required is "3.8")
Call Stack (most recent call first):
  /home/kurth/miniconda3/envs/nest_dev/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /home/kurth/miniconda3/envs/nest_dev/share/cmake-3.21/Modules/FindPython/Support.cmake:3166 (find_package_handle_standard_args)
  /home/kurth/miniconda3/envs/nest_dev/share/cmake-3.21/Modules/FindPython.cmake:514 (include)
  cmake/ProcessOptions.cmake:399 (find_package)
  CMakeLists.txt:130 (nest_process_with_python)


-- Configuring incomplete, errors occurred!

--- a/cmake/ProcessOptions.cmake
+++ b/cmake/ProcessOptions.cmake
@@ -403,6 +403,7 @@ function( NEST_PROCESS_WITH_PYTHON )
       message( WARNING "${PYABI_WARN}")
     else()
       find_package( Python 3.8 REQUIRED Interpreter Development.Module )
+    endif()
 
     if ( Python_FOUND )
       if ( CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT )

--- a/cmake/ProcessOptions.cmake
+++ b/cmake/ProcessOptions.cmake
@@ -392,11 +392,9 @@ function( NEST_PROCESS_WITH_PYTHON )
   elseif ( ${with-python} STREQUAL "ON" )
 
     # Localize the Python interpreter and ABI
-    if ( ${CMAKE_VERSION} VERSION_LESS "3.18.0")
+    find_package( Python 3.8 Interpreter Development.Module )
+    if ( NOT Python_FOUND )
       find_package( Python 3.8 REQUIRED Interpreter Development )
-      message( WARNING "CMake 3.18+ is recommended for more universal Python bindings. If you encounter missing Python header or library file errors, try upgrading CMake.")
-    else()
-      find_package( Python 3.8 REQUIRED Interpreter Development.Module )
     endif()
 
     if ( Python_FOUND )

--- a/cmake/ProcessOptions.cmake
+++ b/cmake/ProcessOptions.cmake
@@ -392,7 +392,7 @@ function( NEST_PROCESS_WITH_PYTHON )
   elseif ( ${with-python} STREQUAL "ON" )
 
     # Localize the Python interpreter and ABI
-    find_package( Python 3.8 Interpreter Development.Module )
+    find_package( Python 3.8 COMPONENTS Interpreter Development.Module )
     if ( NOT Python_FOUND )
       find_package( Python 3.8 REQUIRED Interpreter Development )
       string( CONCAT PYABI_WARN "Could not locate Python ABI"

--- a/cmake/ProcessOptions.cmake
+++ b/cmake/ProcessOptions.cmake
@@ -403,6 +403,8 @@ function( NEST_PROCESS_WITH_PYTHON )
       message( WARNING "${PYABI_WARN}")
     else()
       find_package( Python 3.8 REQUIRED Interpreter Development.Module )
+
+    if ( Python_FOUND )
       if ( CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT )
         execute_process( COMMAND "${Python_EXECUTABLE}" "-c"
           "import sys, os; print(int(bool(os.environ.get('CONDA_DEFAULT_ENV', False)) or (sys.prefix != sys.base_prefix)))"

--- a/cmake/ProcessOptions.cmake
+++ b/cmake/ProcessOptions.cmake
@@ -395,6 +395,13 @@ function( NEST_PROCESS_WITH_PYTHON )
     find_package( Python 3.8 Interpreter Development.Module )
     if ( NOT Python_FOUND )
       find_package( Python 3.8 REQUIRED Interpreter Development )
+      string( CONCAT PYABI_WARN "Could not locate Python ABI"
+        ", using shared libraries and header file instead."
+        " If you encounter missing `libpython.so` or `Python.h` file errors"
+        " please verify that CMake is up-to-date (3.18+) or that Python"
+        " development packages are installed."
+      )
+      message( WARNING "$PYABI_WARN")
     endif()
 
     if ( Python_FOUND )

--- a/cmake/ProcessOptions.cmake
+++ b/cmake/ProcessOptions.cmake
@@ -401,9 +401,8 @@ function( NEST_PROCESS_WITH_PYTHON )
         " is up-to-date (3.18+)."
       )
       message( WARNING "${PYABI_WARN}")
-    endif()
-
-    if ( Python_FOUND )
+    else()
+      find_package( Python 3.8 REQUIRED Interpreter Development.Module )
       if ( CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT )
         execute_process( COMMAND "${Python_EXECUTABLE}" "-c"
           "import sys, os; print(int(bool(os.environ.get('CONDA_DEFAULT_ENV', False)) or (sys.prefix != sys.base_prefix)))"

--- a/cmake/ProcessOptions.cmake
+++ b/cmake/ProcessOptions.cmake
@@ -392,16 +392,15 @@ function( NEST_PROCESS_WITH_PYTHON )
   elseif ( ${with-python} STREQUAL "ON" )
 
     # Localize the Python interpreter and ABI
-    find_package( Python 3.8 COMPONENTS Interpreter Development.Module )
+    find_package( Python 3.8 QUIET COMPONENTS Interpreter Development.Module )
     if ( NOT Python_FOUND )
       find_package( Python 3.8 REQUIRED Interpreter Development )
       string( CONCAT PYABI_WARN "Could not locate Python ABI"
         ", using shared libraries and header file instead."
-        " If you encounter missing `libpython.so` or `Python.h` file errors"
-        " please verify that CMake is up-to-date (3.18+) or that Python"
-        " development packages are installed."
+        " Please clear your CMake cache and build folder and verify that CMake"
+        " is up-to-date (3.18+)."
       )
-      message( WARNING "$PYABI_WARN")
+      message( WARNING "${PYABI_WARN}")
     endif()
 
     if ( Python_FOUND )

